cdap-runtime-ext-dataproc/pom.xml

```diff
@@ -165,6 +165,18 @@
       <groupId>org.powermock</groupId>
       <artifactId>powermock-api-mockito</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-api</artifactId>
+      <version>6.8.0-SNAPSHOT</version>
+      <scope>compile</scope>
+    </dependency>
+      <dependency>
+          <groupId>io.cdap.cdap</groupId>
+          <artifactId>cdap-common</artifactId>
```

Comment:
extensions should maintain the SPI boundary and be treated like they are completely separate from CDAP. In particular, the only CDAP dependency this should have is the cdap-runtime-spi

Reply:
This dependency is brought in because of InvalidConfigurationException which is not specific to extensions i.e. common to CDAP and extensions.  Where should we house such common classes? What is the reason to keep extensions separate from CDAP? Do we expect to use them outside of CDAP?

Reply:
Due to classloading and to avoid incompatible classes, we need to keep ext clean. Instead of exposing the exception for the ext to new it, it could be method that exposed through `ProvisionerContext` to report error.

Reply:
DataprocProvisioner only throws DataprocRuntimeException. So dependency on cdap-common has been removed. But for DataprocRuntimeException to derive from ErrorCodeProvider a dependency on cdap-api is needed. Is this fine?

Reply:
cdap-api is the API for somebody writing a CDAP application. It's better not to pull in all of cdap-api into the cdap-runtime-spi, as it will include a bunch of classes that are not relevant to the runtime spi (it also pulls in gson unfortunately).

Instead, I think it would be better to add some cdap-error-api module that just has these classes and nothing else.

Reply:
Moved ErrorCodeProvider to a new module called cdap-error-api.